Submitted By:            Douglas R. Reno <renodr at linuxfromscratch dot org>
Date:                    2025-05-19
Initial Package Version: 24.2.3.2
Origin:                  Upstream
Upstream Status:         Applied
Description:             Fixes building Libreoffice with poppler-25.05.0
                         by adjusting arguments to be Object::null()
                         instead of Object(objNull). Massive thanks goes to
                         Marty Jack for finding this and putting it into the
                         Libreoffice ticket (#21524)

From 0ee2636304ac049f21415c67e92040f7d6c14d35 Mon Sep 17 00:00:00 2001
From: Xisco Fauli <xiscofauli@libreoffice.org>
Date: Mon, 5 May 2025 10:03:10 +0200
Subject: poppler: uggrade to 25.05

Change-Id: Icd2b318cc01bedc5baeb4445a463a136ed6d7c48
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/184953
Tested-by: Jenkins
Reviewed-by: Xisco Fauli <xiscofauli@libreoffice.org>
---
 sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx |  8 ++++++++
 3 files changed, 17 insertions(+), 9 deletions(-)

diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index dcc1932f7ad7..604ba3b7db2c 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -1317,11 +1317,19 @@ poppler_bool PDFOutDev::tilingPatternFill(GfxState *state, Gfx *, Catalog *,
         }
     }
 
+#if POPPLER_CHECK_VERSION(25, 5, 0)
+    std::unique_ptr<MemStream> pRgbStr(new MemStream(pBitmapData, 0,
+        nBitmapWidth * nBitmapHeight * 3, Object::null()));
+    std::unique_ptr<MemStream> pAlphaStr(new MemStream(reinterpret_cast<char *>(pSplashBitmap->getAlphaPtr()),
+        0, nBitmapWidth * nBitmapHeight, Object::null()));
+    auto aDecode = Object::null();
+#else
     std::unique_ptr<MemStream> pRgbStr(new MemStream(pBitmapData, 0,
         nBitmapWidth * nBitmapHeight * 3, Object(objNull)));
     std::unique_ptr<MemStream> pAlphaStr(new MemStream(reinterpret_cast<char *>(pSplashBitmap->getAlphaPtr()),
         0, nBitmapWidth * nBitmapHeight, Object(objNull)));
     auto aDecode = Object(objNull);
+#endif
 #if POPPLER_CHECK_VERSION(24, 10, 0)
     std::unique_ptr<GfxImageColorMap> pRgbIdentityColorMap(new GfxImageColorMap(8, &aDecode,
         std::make_unique<GfxDeviceRGBColorSpace>()));
-- 
cgit v1.2.3
